linkml_runtime/dumpers/yaml_dumper.py

```diff
@@ -1,18 +1,20 @@
-from decimal import Decimal
-
 import yaml
+from typing import Union
+from pydantic import BaseModel
 
 from linkml_runtime.dumpers.dumper_root import Dumper
 from linkml_runtime.utils.formatutils import remove_empty_items
 from linkml_runtime.utils.yamlutils import YAMLRoot
 
 class YAMLDumper(Dumper):
 
-    def dumps(self, element: YAMLRoot, **kwargs) -> str:
+    def dumps(self, element: Union[BaseModel, YAMLRoot], **kwargs) -> str:
         """ Return element formatted as a YAML string """
         # Internal note: remove_empty_items will also convert Decimals to int/float;
         # this is necessary until https://github.com/yaml/pyyaml/pull/372 is merged
-        return yaml.dump(remove_empty_items(element, hide_protected_keys=True),
+
+        dumper_safe_element = element.dict() if isinstance(element, BaseModel) else element
```

Comment:
Look at that fancy ternary situation, but same deal, element.dict() on a pydantic should make it behave like a YAMLRoot for dumping

Reply:
not totally sure wym about ternary... but this has the side effect of making YAMLRoot dumping more verbose?

Reply:
Good thing to test either way though!